### DISCUSSION TITLE
Setting `user.userName`

### DISCUSSION THREAD

**Discussion by mightyiam** [nix-community/nix-on-droid#297] [Q&A]
It seems that `user.userName` is readonly. But this is the first thing I tried setting.

I'm excited about the home-manager support, but my home-manager config does specify `home.userName`. Will this conflict in any way?

**Comment by t184256** [comment]
IDK, try it, and if it does, lower the priority of it in home-manager config?

**Reply by mightyiam** [reply]
I'm not sure what I'm supposed to do regarding the priority.

The way I worked around this is by avoiding setting the username and home directory.

I'm pretty sure there's a better way forward. Perhaps these options should not be readonly? Would it suffice that they be some high priority? I think that would work for me.

**Reply by t184256** [reply]
> I'm not sure what I'm supposed to do regarding the priority.

`user.userName = lib.mkDefault "desired_username";` in your home-manager config, so that nix-on-droid overrides it.

> Would it suffice that they be some high priority?

No.

> Perhaps these options should not be readonly?

Well-defined suggestions with pull-requests are welcome, though I don't know what's the use in overriding them for an inherently single-user environment.
